File: presentation/src/main/java/com/mbmc/template/ui/RepoAdapter.java
package com.mbmc.template.ui;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;
import com.mbmc.template.BR;
import com.mbmc.template.R;
import com.mbmc.template.domain.entity.Repo;

import java.util.ArrayList;
import java.util.List;

public class RepoAdapter extends RecyclerView.Adapter<RepoAdapter.ViewHolder> {

    private List<Repo> content = new ArrayList<>();

    @NonNull
    @Override
    public RepoAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        LayoutInflater layoutInflater = LayoutInflater.from(viewGroup.getContext());
        return new ViewHolder(DataBindingUtil.inflate(layoutInflater, R.layout.repo_view,
                viewGroup, false));
    }

    @Override
    public void onBindViewHolder(@NonNull RepoAdapter.ViewHolder viewHolder, int i) {
        viewHolder.bind(content.get(i));
    }

    @Override
    public int getItemCount() {
        return content.size();
    }

    public void setContent(List<Repo> content) {
        this.content.clear();
        this.content.addAll(content);
        notifyDataSetChanged();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final ViewDataBinding dataBinding;

        public ViewHolder(ViewDataBinding dataBinding) {
            super(dataBinding.getRoot());
            this.dataBinding = dataBinding;
        }

        public void bind(Repo repo) {
            dataBinding.setVariable(BR.repo, repo);
            dataBinding.executePendingBindings();
        }
    }

}

File: presentation/src/main/kotlin/com/mbmc/template/ui/RepoAdapter.kt
package com.mbmc.template.ui

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.annotation.NonNull
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView
import com.mbmc.template.BR
import com.mbmc.template.R
import com.mbmc.template.entity.Repo

class RepoAdapter : RecyclerView.Adapter<RepoAdapter.ViewHolder>() {
    private val content = ArrayList<Repo>()

    @NonNull
    override fun onCreateViewHolder(@NonNull viewGroup: ViewGroup, position: Int): ViewHolder {
        val layoutInflater = LayoutInflater.from(viewGroup.context)
        return ViewHolder(DataBindingUtil.inflate(layoutInflater, R.layout.repo_view, viewGroup, false))
    }

    override fun onBindViewHolder(@NonNull viewHolder: ViewHolder, position: Int) {
        viewHolder.bind(content[position])
    }

    override fun getItemCount(): Int = content.size

    fun setContent(content: List<Repo>) {
        this.content.clear()
        this.content.addAll(content)
        notifyDataSetChanged()
    }

    inner class ViewHolder(val dataBinding: ViewDataBinding) : RecyclerView.ViewHolder(dataBinding.root) {
        fun bind(repo: Repo) {
            dataBinding.setVariable(BR.repo, repo)
            dataBinding.executePendingBindings()
        }
    }
}
